docs/proposals/logdir-volume.md

```diff
@@ -0,0 +1,128 @@
+<!-- BEGIN MUNGE: UNVERSIONED_WARNING -->
+
+<!-- BEGIN STRIP_FOR_RELEASE -->
+
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+
+<h2>PLEASE NOTE: This document applies to the HEAD of the source tree</h2>
+
+If you are using a released version of Kubernetes, you should
+refer to the docs that go with that version.
+
+<strong>
+The latest 1.0.x release of this document can be found
+[here](http://releases.k8s.io/release-1.0/docs/proposals/logdir-volume.md).
+
+Documentation for other releases can be found at
+[releases.k8s.io](http://releases.k8s.io).
+</strong>
+--
+
+<!-- END STRIP_FOR_RELEASE -->
+
+<!-- END MUNGE: UNVERSIONED_WARNING -->
+
+# LogDir Volume
+
+## Abstract
+
+A proposal for implementing a new type of volume: `LogDir`. A `LogDir` volume can preserve and manage a pod's log files.
+
+Several existing issues and PRs were already created regarding that particular subject:
+* Collect logfiles inside the container [#12567](https://github.com/kubernetes/kubernetes/issues/12567)
+* Add LogDir volume plugin [#13010](https://github.com/kubernetes/kubernetes/pull/13010#issuecomment-133997188)
+
+Author: WuLonghui (@wulonghui)
+
+## Motivation
+
+Some applications will write log to files, we should use a Kubernetes volume
+to preserve the log files, since the vlome supported currently is not very appropriate(refer to [#12567](https://github.com/kubernetes/kubernetes/issues/12567)), so need to add a new type of volume: `LogDir`. It can add some useful infomation to make logging agent easy to collect the logs, also we can manage the log files(like log rotation and maximum log file sizes).
+
+## Implementation
+
+`LogDir` volume is similar to `EmptyDir` volume, so they have the same implementation currently.
+The biggest difference between them is mount point:
+
+```
+* EmptyDir: <kubelet_root_dir>/pods/<pod_id>/volumes/kubernetes.io~empty-dir/<volume_name>
+* LogDir:   <kubelet_root_dir>/pods/<pod_id>/volumes/kubernetes.io~log-dir/<pod_full_name>/<volume_name>
+```
+
+As we can see, `emptyDir` volume add a directory level: `pod_full_name`, the purpose is to make logging agent(e.g.fluentd) easy to add tag by `pod_full_name`.
```

Comment:
I'd like to see how we configure fluentd or other collectors to find these logs.  I'm concerned that the coupling may be too loose (it's a side-effect of the pod spec) and also end up hardcoding things like kubelet root dir (or else add more flags that have to be syncronized across configs).

I wonder if this might make more sense to promote to a full API construct - something like:

```
pod:
    spec:
        containers:
            - name: foo
              logDir: /var/log/foo
        logPolicy:
            rotate: Daily
```

I just threw that out, it's probably not at all the form we would want.